### PR TITLE
DOC 40-Building-on-zySync

### PR DESCRIPTION
**Pull request by krofax**
## Description
Created a section called "Building on zkSync", with some existing sections and added a couple more.
This section puts together the nuggets and "go-to" points for a developer looking to build on building on zkSync.

### CODE REVIEW

File: docs/dev/developer-guides/building-on-zksync/videos.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Developer videos
+
+### zkSync 2.0 for Builders - Alex Gluchowski
+
+This talk covers everything you need to know about zkSync 2.0, and the user, and developer experience perspectives when building on zkSync.
+
+<iframe src="https://www.youtube.com/embed/xd2siik0iBU" title="zSync 2.0 for Builders - Alex Gluchowski" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (uF4No) [inline docs/dev/developer-guides/building-on-zksync/videos.md:7]: Remove the "autoplay;" or all videos would autoplay on load in some browsers. Also try adding a width or height to make this page look better

File: docs/dev/developer-guides/building-on-zksync/tools.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Choosing the right tool
+
+How do you use your favorite tools for building contracts when you're working with Optimism? Just use them!
+
+zkSync EVM compatible, which means all the Ethereum tooling you use, should work on zkSync without much stress.
+
+## Tools available on zkSync
+
+We are currently adding more tools to our belt, to enable developers to build with their favorite tools.
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/tools.md:9]: We are currently adding more tools to our belt. Get in touch through our [Discord](https://discord.com/invite/px2aR7w) if you want to request specific ones.

File: docs/dev/developer-guides/building-on-zksync/tools.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+The following are some tools that you can utilize:
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/tools.md:10]: Remove this line

File: docs/dev/developer-guides/building-on-zksync/tools.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+- Hardhat: Configuring Hardhat to deploy your contracts to zkSync is a breeze. Check out a comprehensive guide on the installation [here](../../../api/hardhat/getting-started.md).
+- Javascript SDK: Building on zkSync on the web with javascript is also pretty a smooth walkthrough, this [guide](../../../api/js/getting-started.md) contains all you need to get started, from creating a wallet to depositing funds, performing a transfer, withdrawing funds and a lot more.
+
+## Other tools
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/tools.md:15]: remove this section

File: docs/dev/developer-guides/building-on-zksync/tools.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Choosing the right tool
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/tools.md:1]: Honestly... I think we should remove this whole page... it is not adding any value

File: docs/dev/developer-guides/building-on-zksync/videos.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+### The zkEVM is Here - Steve Newcomb
+
+In this talk, Steve discusses how zkSync is going to be shipping the world's first zkEVM and how it's going to impact the blockchain space. 
+
+<iframe src="https://www.youtube.com/embed/QkZUlqetTRA" title="The zkEVM is Here - Steve Newcomb" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/videos.md:13]: Antonio made some comments on autoplay and video size.... please review them all

File: docs/dev/developer-guides/security.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
+# Security model
+
+Given that zkSync is still in its early stages of development, there is a chance that our codebase contains undiscovered bugs that could lead to the loss of users' assets. As a result, we are constantly working to strengthen our security protocols to ensure that users and developers have a seamless onboarding experience when building on zkSync.
+zkSync smart contract codebase is presently being audited, but audits are not a guarantee of quality, and they are not a guarantee that a codebase has been thoroughly vetted for vulnerabilities. It's crucial to realize that zkSync usage exposes you to the possibility of defects in the zkSync codebase.
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/security.md:4]: is presently --> is currently

File: docs/dev/developer-guides/security.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+## Bug Bounty Program
+zkSync takes security seriously and as such, we have a massive [bug bounty program](https://immunefi.com/bounty/zksync/). 
+
+## Scope
+Issues that can lead to substantial loss of money, critical bugs like a broken live-ness condition, blocking upgrade system, irreversible loss of funds, or enforced exodus mode.
+
+## Assumptions
+
+To be eligible for the bug bounty, a bug should adhere to the security assumptions of zkSync. You can read more about them [here](https://github.com/matter-labs/zksync/blob/master/docs/protocol.md#assumptions)
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/security.md:14]: Does linking to a v2 doc makes sense?

File: docs/dev/fundamentals/README.md [hunk 9]
```diff
@@ -7,4 +7,5 @@ Learn the basics of zkSync and its ecosystem.
 
 - [Intro to rollups](./rollups.md)
 - [zkSync overview](./zkSync.md)
-- [zkSync testnet](./testnet.md)
\ No newline at end of file
+- [zkSync testnet](./testnet.md)
+- [FAQ](./faq/known-issues.md)
```
Comment (ramonatmatterlabs) [inline docs/dev/fundamentals/README.md:11]: Same comment on renaming the section

File: docs/dev/developer-guides/building-on-zksync/events.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
+# zkSync events
+
+## Overview 
+Given that smart contracts themselves cannot listen to any events, events are a mechanism to publish actions to listeners outside the blockchain.
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/events.md:4]: Events are a mechanism to publish information to listeners outside the blockchain, given that smart contracts themselves can't read them.

File: docs/dev/developer-guides/building-on-zksync/events.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+The blockchain makes all information available to the public, and any actions can be discovered by carefully looking into the transactions, but events are a shortcut for making the development of external systems work with smart contracts; they let Dapps keep track of and respond to what's happening to a smart contract. They can also be searched for because they are indexable. Therefore, you should emit an event anytime something occurs in your smart contract that some system outside the blockchain should be aware of so that the outside system may listen for such occurrences.
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/events.md:6]: Blockchains are public by design, and therefore makes all information available to the public, and any actions can be discovered by carefully looking into the transactions. Events are a shortcut for making specific information easily available for external systems; they let dApps keep track of, and respond to what's happening to a smart contract.

File: docs/dev/developer-guides/building-on-zksync/events.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
+# zkSync events
+
+## Overview 
+Given that smart contracts themselves cannot listen to any events, events are a mechanism to publish actions to listeners outside the blockchain.
+
+The blockchain makes all information available to the public, and any actions can be discovered by carefully looking into the transactions, but events are a shortcut for making the development of external systems work with smart contracts; they let Dapps keep track of and respond to what's happening to a smart contract. They can also be searched for because they are indexable. Therefore, you should emit an event anytime something occurs in your smart contract that some system outside the blockchain should be aware of so that the outside system may listen for such occurrences. 
+Events are included in the transaction logs of the same block that fired the event.
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/events.md:7]: Events are included in the transaction logs of the same block containing the original transaction.

File: docs/dev/developer-guides/building-on-zksync/events.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+zkSync events in terms of their operations are quite similar to Ethereum.
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/events.md:9]: At zkSync, events behave the same way as in Ethereum.

File: docs/dev/developer-guides/building-on-zksync/events.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+//    "method":"eth_getLogs",
+//    "params":[{
+//       "fromBlock": "0x8e8e6",
+//        "toBlock": "0x8e8e6",
+//        "topics": [["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"]]
+//    }]
+//}
+
+```
+zkSync returns a 10K logs limit per response.
```
Comment (ramonatmatterlabs) [inline docs/dev/developer-guides/building-on-zksync/events.md:40]: zkSync has a 10K log limit per response.

File: docs/.vuepress/config.js [hunk 15]
```diff
@@ -63,46 +63,59 @@ module.exports = {
...
           ],
         },
+        {
+          title: "Building on zkSync",
+          path: "/dev/developer-guides",
+          collapsable: false,
+          children: [
+            "/dev/developer-guides/hello-world.md",
+            "/dev/developer-guides/building-on-zksync/tools.md",
+            "/dev/developer-guides/bridging/l1-l2.md",
```
[the remark this answers is not in the document]
Comment (githubdoramon) [reply docs/.vuepress/config.js:104]: The one about l1 - l2 messaging

File: docs/dev/developer-guides/security.md (hunk 8 above)
Comment (krofax) [reply docs/dev/developer-guides/security.md:14]: It's not bad, actually.

Comment (ramonatmatterlabs) [reply docs/dev/developer-guides/security.md:14]: ops... wrong comment.. I meant does linking to a v1 doc makes sense? Is it really valid?

File: docs/dev/developer-guides/building-on-zksync/videos.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
+# Developer videos
+
+### zkSync 2.0 for Builders - Alex Gluchowski
+
+This talk covers everything you need to know about zkSync 2.0, and the user, and developer experience perspectives when building on zkSync.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/xd2siik0iBU" title="zSync 2.0 for Builders - Alex Gluchowski frameborder="0" allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (uF4No) [inline docs/dev/developer-guides/building-on-zksync/videos.md:7]: first video does not load correctly on the page

**Review by uF4No — APPROVED** [review]
All good, just check the videos page cause first video does not show properly

**Review by uF4No — APPROVED** [review]
approved

**Review by ramonatmatterlabs — APPROVED** [review]

**Pull request merged**